experanto/datasets.py

```diff
@@ -438,11 +447,219 @@ def __getitem__(self, idx) -> dict:
+            self._statistics[device_name] = {}
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                mode = self.modality_config[device_name].transforms.normalization
+                assert mode in ['standardize', 'normalize']
+                means = np.load(self._experiment.devices[device_name].root_folder / "meta/means.npy")
+                stds = np.load(self._experiment.devices[device_name].root_folder / "meta/stds.npy")
+                if mode == 'standardize':
+                    means = np.zeros_like(means)
+
+                self._statistics[device_name]["mean"] = means.reshape(1, -1)
+                self._statistics[device_name]["std"] = stds.reshape(1, -1)
 
+    def initialize_transforms(self):
+        transforms = {}
+        for device_name in self.device_names:
+            if device_name == "screen":
+                add_channel = Lambda(lambda x: x[:, None, ...] if isinstance(x, torch.Tensor) and len(x.shape) == 3
+                                 else torch.from_numpy(x[:, None, ...]) if isinstance(x, np.ndarray) and len(x.shape) == 3
+                                 else x)
+                transform_list = [v for v in self.modality_config.screen.transforms.values() if isinstance(v, torch.nn.Module)]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                transform_list = [ToTensor()]
+            if self.modality_config[device_name].transforms.get("normalization", False):
+                transform_list.append(
+                    torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])
+                )
+            transforms[device_name] = Compose(transform_list)
+        return transforms
+
+    def get_swap_time_valid_conditions(self) -> np.ndarray:
+        """
+        Determines valid swap times based on the meta conditions in the modality configuration.
+        """
+        valid_conditions = self.modality_config["screen"]["valid_condition"]
+        include_blanks = self.modality_config["screen"]["include_blanks"]
+
+        valid_indices = np.ones(len(self.meta_conditions["valid_trial"]), dtype=bool)
+
+        for condition_key, condition_value in valid_conditions.items():
+            condition_array = np.array(self.meta_conditions[condition_key])
+            valid_indices &= (condition_array == condition_value) | (
+                (condition_array == "blank") & include_blanks
+            )
+        self._valid_indices = valid_indices
+        # Return the swap times that satisfy all conditions
+        return self._experiment.devices['screen'].timestamps[valid_indices]
+
+
+    def shuffle_valid_screen_times(self) -> None:
+        times = self._full_valid_sample_times
+        self._valid_screen_times = np.sort(np.random.choice(times, size=len(times) // self.sample_stride, replace=False))
```

Comment:
Same here, these seems to be a copypaste from [lines 430-435](https://github.com/sensorium-competition/experanto/blob/33b3c00ce2307dbeef6b63b33596c1e89512e14b/experanto/datasets.py#L430-L435) from the ChunkDataset. I would suggest to move it to a parent class and reuse (please see the comments from the original function)